src/main/java/io/github/opencubicchunks/cubicchunks/mixin/core/common/world/dimension/MixinDimension.java

```diff
@@ -0,0 +1,73 @@
+package io.github.opencubicchunks.cubicchunks.mixin.core.common.world.dimension;
+
+import io.github.opencubicchunks.cubicchunks.CubicChunks;
+import io.github.opencubicchunks.cubicchunks.world.dimension.ICubicDimension;
+import net.minecraft.world.World;
+import net.minecraft.world.dimension.Dimension;
+import net.minecraft.world.dimension.DimensionType;
+import net.minecraftforge.common.extensions.IForgeDimension;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+@Mixin(Dimension.class)
+public abstract class MixinDimension implements IForgeDimension, ICubicDimension {
+
+    private int maxHeight;
+
+    private int minHeight;
+
+
+    @Inject(
+            method = "<init>",
+            at = @At("RETURN")
+    )
+    public void on$init(World world, DimensionType dimensionType, float p_i225788_3_, CallbackInfo ci) {
+
+        // TODO: check GUI override
+
+        // if the ModDimension is an ICubicDimension, use the heights provided by it.
+        if (dimensionType.getModType() instanceof ICubicDimension) {
+            ICubicDimension cubicDimensionType = (ICubicDimension) dimensionType.getModType();
```

Comment:
For now this is fine, but when we add support for vanilla worlds, this should probably also do all the checks for whether it should become a CC world.

I'm also not sure whether this would be the best place for it. The cubic chunks world initialization will have to be done from the scratch anyway and rethought from how it was done in 1.12.2, for now this looks good enough. I hope to not need to do the reinitialization magic I did in 1.12.2.